### ISSUE TITLE
Hesse can un-fix parameters

### ISSUE THREAD

**Issue by JohannesGaessler** [scikit-hep/iminuit#424]
I'm a developer of [kafe2](https://github.com/dsavoiu/kafe2).
iminuit is one of the backends we use.
We noticed that after upgrading iminuit from 1.3.10 to 1.4.2 some of our unit tests for edge cases were failing.
I eventually managed to track down the reason: if you call migrad, then fix one of the parameters, and then call hesse the parameter gets un-fixed.
I assume this is not intended.

Short script that reproduces the behavior:

```python
from iminuit import Minuit


def fcn_3(x, y, z):
    return (x - 1.23)**2 + 0.5*(y - 4.32)**2 + 3*(z - 9.81)**2 - 5.23


m = Minuit(fcn_3, errordef=1.0, x=1.0, y=1.0, z=1.0)
m.migrad()

print(m.matrix(correlation=False, skip_fixed=True))
m.fixed["x"] = True
print(m.fixed["x"])  # True
m.hesse()
print(m.matrix(correlation=False, skip_fixed=True))
print(m.fixed["x"])  # False
```

**Comment by HDembinski** [comment]
Thank you for reporting this, yes, that's a bug, I will look into this, thank you for reporting.

**Comment by HDembinski** [comment]
Fix in preparation here #425

**Comment by HDembinski** [comment]
Fixed in develop

**Comment by HDembinski** [comment]
Now also fixed in release 1.4.3.
